add DAGCircuit.num_qbits() as a replacement for erroneous usage of DA…

jwoehr opened the pull request:
…GCircuit.width() ... First step to implement #2564 providing function to replace inconsistent semantic of width()

<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add it in the CHANGELOG file under Unreleased section.
⚠️ If your pull request fixes an open issue, please link to the issue.

✅ I have added the tests to cover my changes.
✅ I have updated the documentation accordingly.
✅ I have read the CONTRIBUTING document.
-->

### Summary



### Details and comments

ajavadia commented:
can you change `qbits` to `qubits` for consistency with the rest of qiskit?

While you are at it, the `num_cbits` should also probably be changed to `num_clbits` for consistency. E.g. see the Instruction class:
https://github.com/Qiskit/qiskit-terra/blob/24fb5adb79539b2083a096d674cf208a9067bd88/qiskit/circuit/instruction.py#L70

jwoehr commented:
@ajavadia ...

- I renamed `num_qbits()` to `num_qubits()`.
- I added a `num_clbits()`
- `num_cbits()` will need to stay until these changes are merged and invocations or mentions of `num_cbits()` are examined and possibly changed to `num_clbits()` in
  - qiskit-tutorials/qiskit/terra/advanced_circuits.ipynb
  - qiskit-aqua/qiskit/aqua/utils/circuit_utils.py
  - qiskit/docs/terra/custom_gates.rst
